src/components/ColorPicker/components/ColorPickerItemComponent/ColorPickerItemComponent.scss

```diff
@@ -1,101 +1,142 @@
+    &:focus-visible,
+    &.active {
+      outline: none;
+
       .feedback-indicator {
-        @include absolute-border(4px, primary-background-hover-color, -4px, 8px);
+        &:before {
+          @include absolute-border(4px, primary-selected-color);
+        }
+
+        &:after {
+          @include absolute-border(1px, primary-color);
+        }
       }
     }
-  }
 
-  .color-item {
-    display: flex;
-    justify-content: center;
-    align-items: center;
-    border: 2px solid transparent;
-    border-radius: calc(4px * var(--border-radius-multiplier, 1));
-    cursor: pointer;
-
-    .color-indicator-wrapper {
-      margin-top: 3px;
+    &:hover {
+      &:not(.active):not(.selected-color) {
+        // active elements have no "hover feedback"
+        .feedback-indicator {
+          @include absolute-border(4px, primary-background-hover-color, -4px, 8px);
+        }
+      }
     }
 
-    &.color-item-size-small {
-      width: 24px;
-      height: 24px;
-    }
+    .color-item {
+      display: flex;
+      justify-content: center;
+      align-items: center;
+      border: 2px solid transparent;
+      border-radius: calc(4px * var(--border-radius-multiplier, 1));
+      cursor: pointer;
+
+      &.rainbow {
+        background: linear-gradient(
+          0deg,
+          rgba(255, 0, 0, 1) 0%,
+          rgba(255, 154, 0, 1) 10%,
+          rgba(208, 222, 33, 1) 20%,
+          rgba(79, 220, 74, 1) 30%,
+          rgba(63, 218, 216, 1) 40%,
+          rgba(47, 201, 226, 1) 50%,
+          rgba(28, 127, 238, 1) 60%,
+          rgba(95, 21, 242, 1) 70%,
+          rgba(186, 12, 248, 1) 80%,
+          rgba(251, 7, 217, 1) 90%,
+          rgba(255, 0, 0, 1) 100%
+        );
+      }
 
-    &.color-item-size-medium {
-      width: 32px;
-      height: 32px;
-    }
+      .color-indicator-wrapper {
+        margin-top: 3px;
+      }
 
-    &.color-item-size-large {
-      width: 40px;
-      height: 40px;
-    }
-  }
+      &.color-item-size-small {
+        width: 24px;
+        height: 24px;
+      }
 
-  &.selected-color {
-    position: relative;
-    &:before {
-      @include absolute-border(2px, primary-color, -4px);
+      &.color-item-size-medium {
+        width: 32px;
+        height: 32px;
+      }
+
+      &.color-item-size-large {
+        width: 40px;
+        height: 40px;
+      }
     }
 
-    .feedback-indicator {
+    &.selected-color {
+      position: relative;
+
       &:before {
-        // hide the outer border from the "active" feedback indicator
-        display: none;
+        @include absolute-border(2px, primary-color, -4px);
+      }
+
+      .feedback-indicator {
+        &:before {
+          // hide the outer border from the "active" feedback indicator
+          display: none;
+        }
       }
     }
+
+    .color-item-text-mode:hover {
+      background-color: var(--primary-background-hover-color) !important;
+    }
+  }
+
+  .custom-color-picker {
+    transform: translateY(-8);
```

Comment:
should this be ` transform: translateY(-8px);`? 
also, if it looks ok as it is right now, consider just removing this line :)